compiler/rustc_infer/src/traits/util.rs

```diff
@@ -68,59 +68,71 @@ impl<'tcx> Extend<ty::Predicate<'tcx>> for PredicateSet<'tcx> {
 /// `T: Foo`, then we know that `T: 'static`.
 pub struct Elaborator<'tcx> {
     stack: Vec<PredicateObligation<'tcx>>,
-    visited: PredicateSet<'tcx>,
+    seen: PredicateSet<'tcx>,
 }
 
-pub fn elaborate_trait_ref<'tcx>(
-    tcx: TyCtxt<'tcx>,
-    trait_ref: ty::PolyTraitRef<'tcx>,
-) -> Elaborator<'tcx> {
-    elaborate_predicates(tcx, std::iter::once(trait_ref.without_const().to_predicate(tcx)))
+impl<'tcx> Elaborator<'tcx> {
+    // Elaborate the transitive implied predicates of an elaboratable thing `e`.
+    pub fn elaborate(tcx: TyCtxt<'tcx>, e: impl Elaboratable<'tcx>) -> Self {
```

Comment:
Would it make sense to keep the new signatures and names, but make these functions top-level? `Elaborator::elaborate(...)` looks a bit redundant.

Reply:
Or, alternatively, something like `tcx.elaborator().elaborate(elaboratable)`? (given that all these functions create the `Elabarator` first).

Reply:
Hmmmmm... I think I'm gonna keep these names for now. 

I'm planning on adding some more `Elaborator::elaborate_*` names in the future, and it would be nice to have them all "grouped" up under one self type, given how disparate the old imports were (sometimes it was `util::elaborate_*`, sometimes it was `traits::elaborate_*`, sometimes it was just `elaborate_*`, sometimes it was a fully qualified path...)

If you feel very strongly against this naming being redundant, we can discuss.

Reply:
Okay 👍🏻